.circleci/config/base.yml

```diff
@@ -1428,48 +1414,55 @@ commands:
                 condition: << parameters.use-out-cache >>
                 steps:
                   - *step-restore-out-cache
-            - *step-gn-gen-default
-            - *step-electron-build
-            - *step-maybe-electron-dist-strip
-            - step-electron-dist-build:
-                additional-targets: shell_browser_ui_unittests third_party/electron_node:headers third_party/electron_node:overlapped-checker electron:hunspell_dictionaries_zip
-
-            - *step-show-goma-stats
-
-            # mksnapshot
-            - *step-mksnapshot-build
-            - *step-maybe-cross-arch-snapshot
-
-            # chromedriver
-            - *step-electron-chromedriver-build
-
-            - when:
-                condition: << parameters.build-nonproprietary-ffmpeg >>
-                steps:
-                  # ffmpeg
-                  - *step-ffmpeg-gn-gen
-                  - *step-ffmpeg-build
-
-      # Save all data needed for a further tests run.
-      - when:
-          condition: << parameters.persist >>
-          steps:
-            - *step-minimize-workspace-size-from-checkout
-            - run: |
-                rm -rf src/third_party/electron_node/deps/openssl
-                rm -rf src/third_party/electron_node/deps/v8
-            - *step-persist-data-for-tests
 
       - when:
           condition: << parameters.build >>
           steps:
-            - *step-maybe-generate-breakpad-symbols
-            - *step-maybe-zip-symbols
-
-      - when:
-          condition: << parameters.build >>
-          steps:
-            - move_and_store_all_artifacts
+            - build_and_save_artifacts:
+                artifact-key: << parameters.artifact-key >>
+                build-nonproprietary-ffmpeg: << parameters.build-nonproprietary-ffmpeg >>
+            - when:
+                condition:
+                  not:
+                    equal: [ 'nil', << parameters.secondary-artifact-key >> ]
+                steps:
+                  - run:
+                      name: Configuring secondary artifact set
+                      command: |
+                        echo 'export GN_EXTRA_ARGS="<< parameters.secondary-artifact-gn-args >> $GN_EXTRA_ARGS"' >> $BASH_ENV
+                        echo 'export MAS_BUILD="true"' >> $BASH_ENV
+                        rm -rf "src/out/Default/Electron Framework.framework"
+                        rm -rf src/out/Default/Electron*.app
```

Comment:
big yike, can we do this through `ninja -t clean` or something? is there some more kosher way to do this?

Reply:
The issue is that `ninja` doesn't _delete_ output files that aren't expected anymore.  This seems relatively safe and has an almost zero-cost in terms of the build steps

Reply:
the thing i worry about is if there are _other_ files like this. is this the complete list of what we need to delete? i feel like i've done this on my local machine and ended up with a broken build.

Reply:
Discussed in Slack, for now this covers the requisite files.  In the future when build tools has ninja `1.10` we can use `ninja -t cleandead`